### PR TITLE
[WIP] Misc fixes

### PR DESCRIPTION
**Pull request by jasmainak**

### CODE REVIEW

File: scripts/processing/05-run_ica.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def run_ica(subject_id, tsss=None):
     ica.save(ica_name)
 
 
-parallel, run_func, _ = parallel_func(run_ica, n_jobs=N_JOBS)
+# Memory footprint: around n_jobs * 4 GB
+parallel, run_func, _ = parallel_func(run_ica, n_jobs=2)
```
Comment (larsoner) [inline scripts/processing/05-run_ica.py]: Let's just change it to `1` to be safe (users can manually increase as they wish)

Comment (jasmainak) [reply scripts/processing/05-run_ica.py]: okay let me run my ICA scripts, then I change it back :)

File: scripts/processing/05-run_ica.py [hunk 2]
```diff
@@ -32,6 +33,10 @@ def run_ica(subject_id, tsss=None):
         else:
             run_fname = op.join(data_path, 'run_%02d_filt_sss_highpass-%sHz'
                                 '_raw.fif' % (run, 1))
+        if not op.exists(run_fname):
+            warn('Could not find file %s. '
+                 'Skipping run %s for subject %s.' % (run_fname, run, subject))
+            continue
```
Comment (larsoner) [inline scripts/processing/05-run_ica.py]: how did you hit this?

Comment (jasmainak) [reply scripts/processing/05-run_ica.py]: When running the ICA script. Same as in the filtering script, no?

Comment (larsoner) [reply scripts/processing/05-run_ica.py]: Maybe we don't need them in either place, it might be better to raise an error instead. It seems indicative that some previous step was skipped

Comment (jasmainak) [reply scripts/processing/05-run_ica.py]: I thought some files on OpenfMRI were corrupted? According to [this comment](https://github.com/mne-tools/mne-biomag-group-demo/blob/master/scripts/processing/04-python_filtering.py#L41) (probably by Jaakko)

Comment (larsoner) [reply scripts/processing/05-run_ica.py]: I don't remember hitting the problem. Maybe it as an old version of the analysis. I'll try running that script without the `try`

Comment (jasmainak) [reply scripts/processing/05-run_ica.py]: okay let me know. Otherwise, I may have to download the dataset again

**Comment by larsoner** [comment]
FYI I just pushed a minor fix to `master`

https://github.com/mne-tools/mne-biomag-group-demo/commit/9b5f6c44227e6fbe834a8a59304e1216b62ebecb

**Comment by jasmainak** [comment]
okay cool

File: scripts/processing/05-run_ica.py (hunk 2 above)
Comment (larsoner) [reply scripts/processing/05-run_ica.py]: it just ran without error for me with no `try` clause, so let's remove it

File: scripts/processing/00-fetch_data.py [hunk 3]
```diff
@@ -23,7 +24,7 @@
 if not os.path.isdir(archive_dir):
     os.mkdir(archive_dir)
 
-os.systsem('http://openfmri.s3.amazonaws.com/tarballs/ds117_R0.1.1_metadata.tgz')  # noqa: E501
+os.system('http://openfmri.s3.amazonaws.com/tarballs/ds117_R0.1.1_metadata.tgz')  # noqa: E501
```
Comment (agramfort) [inline scripts/processing/00-fetch_data.py]: this really downloads data?

File: scripts/processing/06-make_epochs.py [hunk 4]
```diff
@@ -56,51 +56,51 @@ def run_epochs(subject_id, tsss=False):
...
+        raw = mne.io.read_raw_fif(run_fname, preload=True)
 
         delay = int(round(0.0345 * raw.info['sfreq']))
         events = mne.read_events(op.join(data_path, 'run_%02d-eve.fif' % run))
         events[:, 0] = events[:, 0] + delay
         events_list.append(events)
 
-        raw.info['bads'] = all_bads
+        raw.info['bads'] = bads
+        raw.interpolate_bads()
```
Comment (agramfort) [inline scripts/processing/06-make_epochs.py:88]: @jasmainak why did you change this?

File: scripts/processing/06-make_epochs.py [hunk 5]
```diff
@@ -115,36 +115,40 @@ def run_epochs(subject_id, tsss=False):
...
-        if tsss:
-            ep.average().save(op.join(data_path, '%s-tsss_%d-%s-ave.fif'
-                                      % (subject, tsss, kind)))
-            np.save(op.join(data_path, '%s-tsss_%d-%s-scores.npy'
-                            % (subject, tsss, kind)), sco)
-        else:
-            ep.average().save(op.join(data_path, '%s_highpass-%sHz-%s-ave.fif'
-                                      % (subject, l_freq, kind)))
-            np.save(op.join(data_path, '%s_highpass-%sHz-%s-scores.npy'
-                            % (subject, l_freq, kind)), sco)
```
Comment (agramfort) [inline scripts/processing/06-make_epochs.py:144]: why removing this? @larsoner ok with you?

Comment (jasmainak) [reply scripts/processing/06-make_epochs.py:144]: I didn't remove it. It's still there. Just simplified it a bit. See the variable `filter_label`

File: scripts/processing/06-make_epochs.py (hunk 4 above)
Comment (jasmainak) [reply scripts/processing/06-make_epochs.py:88]: We have bad channels for each run. So, we should interpolate after loading the `raw` for each run rather than after constructing the epochs

**Comment by agramfort** [comment]
I'll try now master branch from scratch on a fresh system

File: scripts/processing/01-anatomy.py [hunk 6]
```diff
@@ -74,7 +74,7 @@ def process_subject_anat(subject_id, force_recon_all=False):
             shutil.copy(f_src, f_dst)
 
     # Fix the headers for subject 19
-    if not op.isfile(op.join(dst_flash, "mef05_0.nii.gz")):
+    if subject_id == 19:
```
Comment (larsoner) [inline scripts/processing/01-anatomy.py:77]: This will make it so this gets re-run every time you run the script, which is not optimal

Comment (jasmainak) [reply scripts/processing/01-anatomy.py:77]: What was there before was not working and this cheap to run anyway, no?

Comment (larsoner) [reply scripts/processing/01-anatomy.py:77]: It's not a lot of overhead, true, but it would have been better to fix the check rather than re-run it. We can leave it though

Comment (jasmainak) [reply scripts/processing/01-anatomy.py:77]: I guess the problem is not missing flash file, but rather a corrupted header. So, you'd have to check the header to know if it was broken or not. But yeah, let's leave it I'd say
